components/delete_category_modal.tsx

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import {ChannelCategory} from 'mattermost-redux/types/channel_categories';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+
+type Props = {
+    category: ChannelCategory;
+    onHide: () => void;
+    deleteCategory: (category: ChannelCategory) => void;
+};
+
+export default class DeleteCategoryModal extends React.PureComponent<Props> {
+    handleCancel = () => {
+        this.props.onHide();
+    }
+
+    handleConfirm = () => {
+        this.props.deleteCategory(this.props.category);
+        this.props.onHide();
+    }
+
+    render() {
+        return (
+            <Modal
```

Comment:
This component shares a lot of structure and CSS with the EditCategoryModal. How much of it could we factor out into a shared component instead of repeating the structure between the two modals? We could use that as the start of a new shared modal component for other things as well since we're going to need it for the What's New modal and such as well.

Reply:
I do plan to create a modal that uses this style that we can use as a common component, but the plan was to tackle that after the main feature was completed.